### PR TITLE
ENH: hook TabularMSA up to pairwise aligners; remove SequenceCollection and Alignment

### LINKED ISSUE #1217: hook TabularMSA up to scikit-bio aligners
Once this is done `SequenceCollection` and `Alignment` can be removed in favor of `TabularMSA`.

### PR DESCRIPTION
**Pull request by jairideout**
Fixes #1217.

### CODE REVIEW

File: skbio/alignment/_pairwise.py [hunk 1]
```diff
@@ -795,9 +838,8 @@ def _init_matrices_nw_no_terminal_gap_penalty(
 
 
 def _compute_substitution_score(aln1_chars, aln2_chars, substitution_matrix,
```
Comment (ebolyen) [inline skbio/alignment/_pairwise.py]: @jairideout @gregcaporaso Is this function ever used?

Comment (ebolyen) [inline skbio/alignment/_pairwise.py]: Nevermind, the order of the helpers was confusing me, the diff made it look recursive on it's own.

Comment (jairideout) [inline skbio/alignment/_pairwise.py]: Yes, it is used in `_compute_score_and_traceback_matrices` which is used by `global_pairwise_align` and `local_pairwise_align`.

**Comment by gregcaporaso** [comment]
Awesome!

**Pull request merged**